src/lib.rs

```diff
@@ -1,7 +1,183 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        assert_eq!(2 + 2, 4);
-    }
+//! YACA - Yet Another Crypto API. Bindings for C library YACA
+//!
+//! Contains two APIs, one objectified a little. And one 1:1 with C
+//! API. First one is available here, second one in module flat.
+
+
+extern crate libc;
+
+
+mod yaca_lib;
+mod yaca_conv;
+
+mod types;
+mod crypto;
+mod simple;
+mod rsa;
+mod key;
+mod digest;
+mod encrypt;
+mod seal;
+mod sign;
+
+
+pub mod prelude {
+    pub use crate::crypto::ContextWithPadding;
+    pub use crate::crypto::ContextWithRc2Supported;
+    pub use crate::crypto::ContextWithXcmEncryptProperties;
+    pub use crate::crypto::ContextWithXcmDecryptProperties;
+}
+
+// ******************************
+// *   (mostly) objective API   *
+// *    use yaca;               *
+// *    use yaca::prelude::*;   *
+// ******************************
+pub type Result<T> = std::result::Result<T, Error>;
+
+pub use prelude::*;
+
+pub use types::Error;
+pub use types::KeyFormat;
+pub use types::KeyFileFormat;
+pub use types::KeyType;
+pub use types::KeySubType::{self, *};
+pub use types::KeyLength;
+pub use types::KeyLengthEc::{self, *};
+pub use types::KeyLengthDh::{self, *};
+pub use types::DigestAlgorithm;
+pub use types::EncryptAlgorithm;
+pub use types::BlockCipherMode;
+pub use types::Padding;
+pub use types::Kdf;
```

Comment:
I was adding them one by one as they were being implemented. I think I'll leave as it is now. Same can be said about crypto and simple in case of object API. Maybe we'll come back here when the API is stabilized.

Reply:
Sure, no problem.

Reply:
I've added TODO there to reconsider in the future.